pkg/controller/service/servicecontroller.go

```diff
@@ -316,15 +312,40 @@ func (s *ServiceController) createLoadBalancerIfNeeded(namespacedName types.Name
 	} else {
 		glog.V(2).Infof("Ensuring LB for service %s", namespacedName)
 
-		// TODO: We could do a dry-run here if wanted to avoid the spurious cloud-calls & events when we restart
-
-		// The load balancer doesn't exist yet, so create it.
-		s.eventRecorder.Event(service, api.EventTypeNormal, "CreatingLoadBalancer", "Creating load balancer")
-		err := s.createLoadBalancer(service)
-		if err != nil {
-			return fmt.Errorf("Failed to create load balancer for service %s: %v", namespacedName, err), retryable
+		// Check if there is old LB there and if it need to be update
+		// Check if the balancer has already existed
+		// Avoid the spurious cloud-calls & events when we restart
+		skipCreate := false
+		if appliedState != nil {
+			// We should delete the old LB if the old LB does not
+			// match what we need now
```

Comment:
You shouldn't need to delete the loadbalancer if it is misconfigured, EnsureTCPPLoadBalancer (called from createLoadBalancer) should fix the broken things in a verification loop (https://github.com/kubernetes/kubernetes/blob/fdc433a9ee54e6a6811d76a3045cb28fe8305fd2/pkg/cloudprovider/providers/gce/gce.go#L378).